tests/test_validators.py

```diff
@@ -736,6 +736,72 @@ def check_something(cls, v):
     assert val_calls == [None, 'hello']
 
 
+def test_required_optional():
```

Comment:
Can you move this to another field, don't really mind where it goes but definitely shouldn't be in this file which is about validators.

Reply:
maybe `test_edge_cases.py` for now, we could probably do with rearranging tests one day.

Reply:
Sure.